screeps-game-api/src/objects/mod.rs

```diff
@@ -147,9 +185,6 @@ pub unsafe trait RoomObjectProperties:
 /// The reference returned by `AsRef<Reference>::as_ref` must reference a
 /// JavaScript object extending the `Structure` class.
 pub unsafe trait StructureProperties: RoomObjectProperties {
```

Comment:
Since we don't have the blanket impl for all structures anymore, I think it would make sense to add `+ HasId` as a requirement for `StructureProperties` here.

Just to let someone require `T: StructuctureProperties` and know that `T` always has an ID as well.